src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -807,6 +807,16 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 				listener.dispose();
 			});
 		}));
+		this._register(dom.addDisposableListener(xterm.element, 'touchstart', () => {
+			xterm.focus();
+			this._terminalFocusContextKey.set(true);
+			if (this.shellType) {
+				this._terminalShellTypeContextKey.set(this.shellType.toString());
+			} else {
+				this._terminalShellTypeContextKey.reset();
+			}
+			this._onDidFocus.fire(this);
```

Comment:
I think this should happen as a result of `xterm.focus()`? I checked and that is the call that happens when clicked:

https://github.com/microsoft/vscode/blob/802a170eff10f49641cb4229b2ac719723eba33b/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts#L818-L826

Reply:
yes but that doesn't happen on touch. so maybe an upstream xterm.js issue?

Reply:
https://github.com/microsoft/vscode/pull/129128/files#diff-8df911bf58ca35baec7e0c02b60c36c1dc33694f7a61089ba69adf888c98f466R812-818